Automattic/jetpack
# Licensing: Add conditional Protect redirect post Scan license activation

## Description
After a Scan license is activated in My Jetpack, the `ActivationScreenSuccessInfo` component displays a link to `View scan results` in Jetpack Cloud. As part of the standalone first initiative, when we detect that Jetpack Protect is active, we should always direct users to the Protect UI to view these results.

## Proposed changes:

* Rearranges the current links
* Adds conditional logic to update the `View scan results` redirect to Cloud or Protect when applicable

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1365

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Create a new JN instance using the beta tester with Jetpack installed using this branch
* Purchase a Scan license from `jetpack.com` but don't yet assign it to a site
* Refer to Store Admin for your user and locate the license key
* Proceed to the JN site and to `/wp-admin/admin.php?page=my-jetpack#/add-license`
* Enter key and select activate, from the post activation screen verify that the UI display as in the provided screenshot
* Ensure that the `View scan results` link redirects to Scan's Cloud dashboard and that the `View my plans` link performs as expected
* Perform the same steps once again but this time after first installing and activating Protect
* Verify that the `View scan results` link now redirect to `/wp-admin/admin.php?page=jetpack-protect#/scan`
* Ensure no regressions are introduced and that other license activations are unaffected by the changes

## Screenshot
![Screen Shot on 2024-08-21 at 14-29-40](https://github.com/user-attachments/assets/f1ab63cc-27ac-4725-bf9d-47bfc178a2e1)
